### ISSUE TITLE
Keys with array indexes don't work

### ISSUE THREAD

**Issue by johnkingsley** [ngx-formly/ngx-formly#1584]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request
```

**Current behavior**

Keys with array indexes don't work properly.
For example, a key of "myarr.0.value" does not create a 'myarr' as an array containing an object with property of "value"

**Expected behavior**

Keys should create arrays as needed.
For example, a key of "myarr.0.value" would create the following in the data model:
```
{
  "myarr": [
      {
         "value": "the value is saved here"
      }
  ]
}
```

**Minimal reproduction of the problem with instructions**

https://stackblitz.com/edit/angular-a3etwt

**What is the motivation / use case for changing the behavior?**

Would like to save data in arrays.

**Please tell us about your environment:**
* **Operating system:** Linux
* **Angular version:** 7.2.15
* **Browser:** all
* **Language:** all
* **Node (for AoT issues):** `node --version` = n/a

**Comment by johnkingsley** [comment]
The problem seems to be in the assignModelValue function in core/src/lib/utils.ts
Assuming this is accepted as a bug, I can submit a pull request to fix it.

**Comment by aitboudad** [comment]
that's a regression, PR welcomed!

**Comment by aitboudad** [comment]
I'll take care of it.

**Comment by aitboudad** [comment]
fixed in `5.1.4`

**Comment by johnkingsley** [comment]
Thanks for fixing it so quickly.  I tested it and all is good.
